sys/aarch64/trap.c

```diff
@@ -153,15 +111,19 @@ void user_trap_handler(mcontext_t *uctx) {
 void kern_trap_handler(ctx_t *ctx) {
   register_t esr = READ_SPECIALREG(esr_el1);
   register_t far = READ_SPECIALREG(far_el1);
+  register_t exc_code = ESR_ELx_EXCEPTION(esr);
 
   /* If interrupts were enabled before we trapped, then turn them on here. */
   if ((_REG(ctx, SPSR) & PSR_I) == 0)
     cpu_intr_enable();
 
-  switch (ESR_ELx_EXCEPTION(esr)) {
+  switch (exc_code) {
     case EXCP_INSN_ABORT:
     case EXCP_DATA_ABORT:
-      abort_handler(ctx, esr, far, false);
+      klog("%x at $%lx, caused by reference to $%lx!", exc_code, _REG(ctx, PC),
+           far);
```

Comment:
This log is common for all architectures - maybe it's good idea to move that to `pmap_fault_handler`? WDYT?

Reply:
I thought about this. The problem is that on RISC-V and MIPS we print an exception string instead of a bare exception code (`exc_code`). Each architecture would have to provide a function like `const char *exc_str(u_long exc_code)` and if the function was missing, a bare exception code would be printed. I discussed this solution with @cahirwpz and we came to conclusion that leaving the printing to the MD module is a better solution.